src/anonymousUserTracking/criteriaCompletionChecker.ts

```diff
@@ -346,6 +346,25 @@ class CriteriaCompletionChecker {
       }
 
       if (field.includes('.')) {
+        const fields = field.split('.');
```

Comment:
Why are we only getting the first `field` value? Shouldn't we run this compute on all `fields`?

Reply:
as we have to take data from the array that's why we are taking only first value.

for example we have field `furniture.furnitureColor` and we have `furniture` as an array from the frontend,
so we have to check that `furniture` is an array or not, if it is an array then we are checking the each element of an array that matches or not